src/components/FileImportModal/FileImportModal.tsx

```diff
@@ -1,48 +1,51 @@
-import React, { FC, useCallback, useRef, memo, FormEvent, useState } from 'react';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faFileImport, faTimes } from '@fortawesome/free-solid-svg-icons';
+import React, { FC, useState, useRef, FormEvent, useCallback, memo } from 'react';
 import {
   Button,
   Dialog,
-  DialogTitle,
-  DialogContent,
   DialogActions,
-  Input,
-  IconButton
+  DialogContent,
+  DialogTitle,
+  IconButton,
+  Input
 } from '@material-ui/core';
-
-import { useCriteriaContext } from 'components/CriteriaProvider';
 import useStyles from './styles';
+import { faFileImport, faTimes } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 
-interface ImportCriteriaModalProps {
+interface FileImportModalProps {
   open: boolean;
   onClose: () => void;
+  onSelectFile: (files: FileList | undefined | null) => void;
+  allowedFileType?: string; // Setting this to undefined will allow any file type to be selected
 }
 
-const ImportCriteriaModal: FC<ImportCriteriaModalProps> = ({ open, onClose }) => {
+const FileImportModal: FC<FileImportModalProps> = ({
+  open,
+  onClose,
+  onSelectFile,
+  allowedFileType
+}) => {
   const styles = useStyles();
   const [fileName, setFileName] = useState<string>('');
   const importFileRef = useRef<HTMLInputElement>(null);
 
-  const { addCriteria } = useCriteriaContext();
-
   const selectFile = useCallback(
     (event: FormEvent<HTMLFormElement>) => {
       event.preventDefault();
       const files = importFileRef?.current?.files;
-      if (files?.length) addCriteria(files[0]);
+      onSelectFile(files);
       onClose();
       setFileName('');
     },
-    [addCriteria, onClose]
+    [onClose, onSelectFile]
   );
 
   const handleChooseFile = useCallback(() => {
     if (importFileRef?.current?.files?.[0]) setFileName(importFileRef.current.files[0].name);
   }, [importFileRef]);
 
   return (
-    <Dialog open={open} onClose={onClose} aria-labelledby="import-criteria" fullWidth maxWidth="sm">
+    <Dialog open={open} onClose={onClose} aira-labelledby="import-pathway" fullWidth maxWidth="sm">
```

Comment:
Somehow a couple typos got introduced here.... `aria` -> `aira` here and `types` -> `tpyes` a few lines below

Reply:
oops